### ISSUE TITLE
Connect to Server Dialog Partly Broken?

### ISSUE THREAD

**Issue by geoffthemedio** [freeorion/freeorion#1417]
This has been the case for ever, at least for me, but the "Lan game at" list in the server connect dialog never seems to do anything. I'm not sure what it's supposed to do exactly, but always be a big empty black square probably wasn't the intention.

![connect_to_server](https://cloud.githubusercontent.com/assets/11386908/24081431/7c604d14-0cb3-11e7-8b4b-f19fdeea8b46.png)

**Comment by adrianbroher** [comment]
I assume this list should show any running server in the local network via TCP broadcast.  You may remember that a server starts a 'discovery' server and a 'game' server.  The client does send a broadcast message to the discovery port and any running 'discovery' server will respond to this broadcase message with something like a 'I'm here at this IP' answer.

**Comment by geoffthemedio** [comment]
Whether localhost, LAN, or remote IP, it would ideally

**Comment by Vezzra** [comment]
> I think it could be also used to show multiplayer servers from metaserver like it done in freeciv:

Well, for that we need to implement and set up a proper metaserver first. Then we can add such functionality to the FO client, sure.

**Comment by Dilvish-fo** [comment]
> Well, for that we need to implement and set up a proper metaserver first. 

I don't think that _necessarily_ needs to come first-- we could simply have a ```servers.inf``` file in the ```default/customizations``` folder

**Comment by Vezzra** [comment]
Isn't the idea of a metaserver basically to have a central registry of publicly accessible game servers, where you can publish your public server if you want to provide one? How could you achieve that just with a local file...?

**Comment by Dilvish-fo** [comment]
> How could you achieve that just with a local file...?

I'm saying it is not necessary to achieve quite all that to get significant usefulness from o0leg's suggestion.  We could prepopulate the file with the only currently known public server (o0leg's) and perhaps the multiplayer group would like their server address included as well.  Or perhaps they, or other groups in their situation, would rather just manually edit the server.inf file to include their server IP address, to make their server easier to access while keeping it semi-private.   As more public servers started up we could just periodically update the server.inf list.  Having a metaserver to handle that kind of thing (and do dynamic checking of which ones are actually running at any given time) would be nice, sure, but hardly seems truly necessary when the total number of servers can be counted on one hand.

**Comment by Vezzra** [comment]
Agreed. Pulling a list of game servers from a local file shouldn't be too hard to implement, and would certainly be a nice feature. I still think a metaserver (like the freeciv example @o01eg gave) is a different story, but you're right of course, currently that would be quite the overkill anyway.

Thinking more about it, it also shouldn't be too hard to pull a list of publicly accessible game servers from a file hosted on a remote site. For example, we could host and maintain such a file in our github repo, shouldn't be too hard to pull it from there? We can even combine the contents of that public file with a local file, where players can maintain their own list of servers.

"Registering" your game server with the public list, if you want to provide one, could then be done e.g. by providing a simple PR. 😉
